src/walk.rs

```diff
@@ -420,7 +420,11 @@ fn spawn_senders(
                                 .ok()
                                 .map_or(false, |m| m.file_type().is_symlink()) =>
                     {
-                        DirEntry::broken_symlink(path)
+                        // The documentation for ignore::Error::depth says that it returns the
+                        // depth "if this error was generated from a recursive directory iterator",
+                        // which it is here.
+                        let depth = inner_err.depth().unwrap();
```

Comment:
Unfortunately, it does not work and panics because ignore creates a variant `Error::Io` for broken symlinks which does not have depth (`None` is returned).